src/Codeception/Module/REST.php

```diff
@@ -333,6 +333,18 @@ public function amNTLMAuthenticated($username, $password)
     }
 
     /**
+     * Adds AWS Authentication via key, secret, service and region
+     * @param $key
+     * @param $secret
+     * @param $service
+     * @param $region
+     */
+    public function amAWSAuthenticated($key, $secret, $service, $region)
+    {
+        $this->client->setAwsAuth($key, $secret, $service, $region);
```

Comment:
I don't think that `key` and especially `secret` should be visible and hardcoded in tests.
In the current implementation, you will need to pass them directly. Which is not a very secure way. 
I think this should be moved to configuration section, so this values could be loaded from environment variables or external config files.

I'd add to REST config options like:

```
aws:
    key:
    secret: 
    service:
    region:
```
any of those options could be overridden inside `amAWSAuthenticated`. So if you need to pass requests to a different region or different service you could set it there.

```php
public function amAWSAuthenticated($addiotionalAWSConfig = [])
{
    $config = array_merge($this->config['aws'], $addiotionalAWSConfig);
}
```

Reply:
Haha I haD initially coded it to be passed from Config files, but then saw none of other authorizations were following that. So decided to hard code it. Will be pushing the fix in few hours.

Reply:
Done

Reply:
> Haha I haD initially coded it to be passed from Config files, but then saw none of other authorizations were following that. So decided to hard code it. Will be pushing the fix in few hours.

yeah, probably the difference here is that other Auth methods can be executed locally. In this case you don't have strong security restrictions. However, for AWS you are providing your AWS account credentials, and they should never appear in tests (and test log)